lib/block-supports/background.php

```diff
@@ -30,6 +30,37 @@ function gutenberg_register_background_support( $block_type ) {
 	}
 }
 
+/**
+ * Given a theme.json or block background styles, returns the background styles for a block.
+ *
+ * @param  array  $background_styles Background style properties.
+ * @return array                     Style engine array of CSS string and style declarations.
+ */
+function gutenberg_get_background_support_styles( $background_styles = array() ) {
+	$background_image_source             = isset( $background_styles['backgroundImage']['source'] ) ? $background_styles['backgroundImage']['source'] : null;
+	$background_styles['backgroundSize'] = ! empty( $background_styles['backgroundSize'] ) ? $background_styles['backgroundSize'] : 'cover';
+
+	/*
+	 * @TODO document
+	 * $background_styles['backgroundImage']['url'] and file source implies an image URL path.
```

Comment:
This needs to be documented. The logic, I believe, is as follow:

1. backgroundImage is a string: We accept whatever the value is. It could be `url(...)` or some combination thereof. The style engine will sanitize.
2. backgroundImage is an object with url and source props: it will be processed by Gutenberg, in the case of "file" or "theme" wrapped in a `url()` function and escaped. CSS defaults like `cover` will kick in.